nodeup/pkg/model/context.go

```diff
@@ -211,6 +211,17 @@ func (c *NodeupModelContext) UseEtcdTLS() bool {
 	return false
 }
 
+// UseTLSAuth checks the peer-auth is set in both cluster
+// @NOTE: in retrospect i think we should have consolidated the common config in the wrapper struct; it
+// feels wierd we set things like version, tls etc per cluster since they both have to be the same.
+func (c *NodeupModelContext) UseTLSAuth() bool {
+	if len(c.Cluster.Spec.EtcdClusters) != 2 {
```

Comment:
For now, I'd probably just loop through and return true if any of the have EnableTLSAuth set.  I don't think mixing & matching will be very common, and I'd rather err on the side of caution.

I'm thinking about splitting out the "etcd-controller" out of protokube as well, and then you will be able to have full options here.  But it's just an inkling of an idea at the moment

Reply:
good call .. updating